signac/contrib/project.py

```diff
@@ -2193,17 +2158,15 @@ def _repr_html_(self):
         return repr(self) + self._repr_html_jobs()
 
 
-def init_project(name, root=None, workspace=None, make_dir=True):
-    """Initialize a project with the given name.
+def init_project(root=None, workspace=None, make_dir=True):
```

Comment:
Since we're already breaking this API, I might suggest requiring keyword arguments here. We're going to have lots of signac 1.x users get confused when they try to run `signac.init_project(name)` and `name` is interpreted as a root directory. Another option would be to catch positional arguments as `*args` and raise a user-friendly error if any are provided, which could help guide users towards 2.x syntax.
```suggestion
def init_project(*, root=None, workspace=None, make_dir=True):
```

Reply:
Yeah this is going to be a problem. Unlike with the `doc_filter` parameter I mentioned above, we can't provide backwards compatibility layers here because `name` is a positional parameter and it comes first in the list. I cannot think of any solution to that. Making the other parameters keyword-only is a reasonable alternative since I suspect that very few people bother to customize any of the other parameters. We could always remove that behavior in 3.0 after users have had sufficient time to accustom themselves to the new behavior.

Reply:
Agreed, this one will be tricky, especially because this function is used [everywhere](https://github.com/search?p=2&q=signac.init_project&type=Code).

Maybe, we could be even more gentle and just raise a deprecation warning when the function is called with only one argument and ignore it?

Reply:
So this:
```
def init_project(*args, root=None, workspace=None, make_dir=True):
    if args:
        if len(args) == 1:
            warnings.warn(f"Project names were removed in signac 2.0. If you intended to call `init_project` with a root directory, please provided it as a keyword argument: `init_project(root={args[0]})`")
        else:
            # Match the usual error from misusing keyword-only args
            raise TypeError(f"init_project() takes 0 positional arguments but {len(args)} was given")

Reply:
Yes, I think so. We might want to consider the warning type.

Reply:
That seems like a good solution to me. The warning type should probably be a FutureWarning since this is effectively a deprecation that we're pushing to 3.0.